### PR TITLE
Added reset simulation tutorial

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: Alejandro Hernández Cordero <ahcorde@gmail.com>

# 🎉 New feature

## Summary
Added reset simulation tutorial

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by mjcarroll** [comment]
Friendly ping @ahcorde, there are some outstanding comments here.

File: tutorials/reset_simulation.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+\page reset_simulation Reset simulation
+
+The Reset Gazebo transport API is exposed to allow resetting simulation to time zero.
+It's possible to call this API using the command line or through the GUI.
+In addition to the API, we have also expanded the simulation system API with a Reset interface.
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply tutorials/reset_simulation.md:5]: @mjcarroll, when you implemented this feature, do you consider any limitations ?

[the remark this answers is not in the document]
Comment (mjcarroll) [reply tutorials/reset_simulation.md:5]: I don't believe that there are any major limitations with systems implementing the reset interface.  They should have full access to the ECM functionality.

Systems that you may want to implement reset for are typically things that have a lot of loaded assets.  This prevents having to reload those assets, and instead allow for a "warm" reboot of the system.

**Review by mabelzhang — APPROVED** [review]
Have a good break!

**Pull request merged**